packages/create-pantheon-decoupled-kit/src/templates/next-drupal-app/src/middleware.ts

```diff
@@ -0,0 +1,58 @@
+import { match as matchLocale } from '@formatjs/intl-localematcher';
+import Negotiator from 'negotiator';
+import type { NextRequest } from 'next/server';
+import { NextResponse } from 'next/server';
+import getLocales from '../scripts/get-locales';
```

Comment:
Ideally we would get the locales dynamically and wouldn't need to set the at all in the `next.config.js`. Additionally it would be nice if we had no js left and only typescript.

1. Refactor `get-locales.js` into a typescript function in `src/lib` with the same functionality.
2. Remove the locale setting in `next.config.js`
3. in `stores.ts`, use the new function to get the value of locales to pass in to `makeLocaleStores`
4. also update this file with the new function